src/widgets/custom_text_dialog.rs

```diff
@@ -92,6 +95,36 @@ mod imp {
                 .sync_create()
                 .build();
 
+            let open_button = self.open_button.get();
+            open_button.connect_clicked(glib::clone!(
+                #[weak(rename_to = imp)]
+                self,
+                move |_| {
+                    let dialog = gtk::FileDialog::new();
+                    let obj = imp.obj();
+
+                    let filter = gtk::FileFilter::new();
+                    filter.set_name(Some("Text Files"));
```

Comment:
This needs to be made translatable!

```suggestion
                    filter.set_name(Some(&gettext("Text Files")));
```